File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        mp = {}
        
        for c in magazine:
            mp[c] = 1+mp.get(c,0)
            
        for c in ransomNote:
            if c not in mp:
                return False
            
            else:
                mp[c]-=1
                
                if mp[c]==0:
                    del mp[c]
                    
        return True

File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string r, string m) {
        unordered_map<char,int> mp;
        
        for(auto x: m){
            if(mp.find(x) == mp.end()){
                mp[x]=1;
            }
            else
                mp[x]+=1;
        }
        
        for(auto x:r){
            if(mp.find(x)!=mp.end()){
                mp[x]-=1;
                if(mp[x]==0)
                    mp.erase(x);
            }
            else
                return false;
        }
        return true;
    }
};